pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -50,6 +50,12 @@ func ShouldPodBeInEndpoints(pod *v1.Pod) bool {
 	return true
 }
 
+// ShouldSetHostname returns true if the Hostname attribute should be set on an
+// Endpoints Address or EndpointSlice Endpoint.
+func ShouldSetHostname(pod *v1.Pod, svc *v1.Service) bool {
```

Comment:
nit: a unit test for this?

to prevent regression.

Reply:
Good call, added a unit test covering this function now.